### PR TITLE
Clear product.variants.unit_description if product.variant_unit is items

### LINKED ISSUE #1582: Unit description not cleared when switching between kg and items
Picked up in #1452. When switching variant_unit between `weight` products and `items`, the unit description text on variants lingers. It should to be removed.

issue1
issue2

## Severity
S3, not affecting so many users I guess and there is a workaround.

### PR DESCRIPTION
**Pull request by luisramos0**
When product.variant_unit is items, the product.variant_unit_name is used instead of the variant.unit_description (this field is not available in the variants edit page if product.variant_unit is items). So, we need to clear it.

#### What? Why?

Closes #1582

#### What should we test?
The product edit page needs to be tested as well as verifying the issue if fixed.

#### Release notes
Changelog Category: Fixed
Unit description (only used for weight or volume products) is cleared when switching a product unit from weight or volume to items.

### CODE REVIEW

File: spec/controllers/spree/admin/products_controller_spec.rb [hunk 1]
```diff
@@ -150,17 +150,31 @@
   end
 
   describe "updating" do
+    let(:producer) { create(:enterprise) }
+    let!(:product) { create(:simple_product, supplier: producer) }
+
+    before do
+      @request.env['HTTP_REFERER'] = 'http://test.com/'
```
Comment (Matt-Yorkley) [inline spec/controllers/spree/admin/products_controller_spec.rb]: Why is this needed here?

Comment (mkllnk) [reply spec/controllers/spree/admin/products_controller_spec.rb]: Yeah, it doesn't seem to belong here. :man_shrugging:

Comment (luisramos0) [reply spec/controllers/spree/admin/products_controller_spec.rb]: must have been a copy paste error... removed in new commit.

**Comment by RachL** [comment]
So I tested with the follwoing super-real scenario:
1 bag of 1g of carrots:
image

After moving to item:
image

It also displays correctly in inventory. Editing other fields still work as well as checkout and BOM.
An old issue fixed!!!
